packages/rum-core/src/common/patching/event-target-patch.js

```diff
@@ -0,0 +1,185 @@
+  }
+
+  let proto = window.EventTarget.prototype
+  const nativeAddEventListener = proto[ADD_EVENT_LISTENER_STR]
+  const nativeRemoveEventListener = proto[REMOVE_EVENT_LISTENER_STR]
+
+  function findTaskIndex(existingTasks, eventType, listenerFn, capture) {
+    for (let i = 0; i < existingTasks.length; i++) {
+      const task = existingTasks[i]
+      if (
+        task.eventType === eventType &&
+        task.listenerFn === listenerFn &&
+        task.capture === capture
+      ) {
+        return i
+      }
+    }
+    return -1
+  }
+
+  function isCapture(options) {
+    let capture
+    if (options === undefined) {
+      capture = false
+    } else if (options === true) {
+      capture = true
+    } else if (options === false) {
+      capture = false
+    } else {
+      capture = options ? !!options.capture : false
+    }
+    return capture
+  }
+
+  function createListenerWrapper(target, eventType, listenerFn, options) {
+    const eventSymbol = eventTypeSymbols[eventType]
+
+    if (!eventSymbol) return listenerFn
+
+    let existingTasks = target[eventSymbol]
+
+    let capture = isCapture(options)
+
+    if (existingTasks) {
+      let taskIndex = findTaskIndex(
+        existingTasks,
+        eventType,
+        listenerFn,
+        capture
+      )
+      if (taskIndex !== -1) {
+        let task = existingTasks[taskIndex]
+        return task.wrappingFn
+      }
+    } else {
+      existingTasks = target[eventSymbol] = []
+    }
+
+    let task = {
+      source: EVENT_TARGET,
+      target,
+      eventType,
+      listenerFn,
+      capture,
+      wrappingFn
+    }
+
+    existingTasks.push(task)
+
+    function wrappingFn(event) {
+      task.event = event
+      callback(SCHEDULE, task)
+      let result
+      try {
+        result = listenerFn.apply(this, arguments)
+      } finally {
+        callback(INVOKE, task)
+      }
+      return result
+    }
+
+    return wrappingFn
+  }
+
+  proto[ADD_EVENT_LISTENER_STR] = function(
+    eventType,
+    listenerFn,
+    optionsOrCapture
+  ) {
+    if (eventTypes.indexOf(eventType) < 0 || typeof listenerFn !== 'function') {
+      return nativeAddEventListener.apply(this, arguments)
+    }
+    const target = this
+
+    const wrappingListenerFn = createListenerWrapper(
+      target,
+      eventType,
+      listenerFn,
+      optionsOrCapture
+    )
+
+    let args = [...arguments]
+    args[1] = wrappingListenerFn
+
+    return nativeAddEventListener.apply(target, args)
+  }
+
+  proto[REMOVE_EVENT_LISTENER_STR] = function(
+    eventType,
+    listenerFn,
+    optionsOrCapture
+  ) {
+    if (eventTypes.indexOf(eventType) < 0 || typeof listenerFn !== 'function') {
+      return nativeRemoveEventListener.apply(this, arguments)
+    }
+    const target = this
+
+    const eventSymbol = eventTypeSymbols[eventType]
+
+    let existingTasks = target[eventSymbol]
+    let capture = isCapture(optionsOrCapture)
+    let args = [...arguments]
+    if (existingTasks) {
```

Comment:
existingTasks.length > 0

Reply:
This only checks if the `existingTasks` is not undefined otherwise if there are no elements it won't find the wrapping function.

Reply:
I understand that part, but this would either be undefined || [ ] for the first time. So instead of just checking for undefined, you could do both `existingTasks && existingTasks.length > 0` and that would avoid unnecessary function calls like `isCapture` and `findTaskIndex`. 

Does it make sense?